src/gui_common/CustomPopupMenu.cs

```diff
@@ -57,8 +90,8 @@ protected override void OnOpen()
     protected override void OnClose()
     {
         var tween = CreateTween();
-        tween.TweenProperty(this, "rect_size", Vector2.Zero, 0.15f)
-            .From(cachedMinSize)
+        tween.TweenProperty(this, "rect_scale", Vector2.Zero, 0.15f)
```

Comment:
Why is the scale now tweened instead of the size?

Reply:
I can't entirely remember if there was a functional reason to change the way this works or if the old way broke stuff. All I can remember is I borrowed it from the CustomWindow's way of tweening for animations. I'll experiment and look into it to see if this really was a necessary change tomorrow, and if not revert it if that's what you'd prefer.

Reply:
I did a quick playtest, and this definitely makes the popups look much different. Before they rolled down from the top to display themselves. Now it looks like the whole popup just shrinks down to the top left corner. So the animation was completely changed and it kind of looks like a cheaper effect in my eyes.

Reply:
Thinking about it more, I think I made this change because the changing of size that was there before broke the moment the minimum size wasn't zero, which it wasn't in earlier versions where I experimented with changing minimum size. So can probably be safely reverted presuming the CustomPopupMenu insures the minimum size is always zero.

Reply:
Do you want this removed / reverted?

Reply:
Is it possible to switch back to the old size based animation? If so, I'd probably prefer that over the scale based animation.

Reply:
Probably possible? I don't think the way the pop-ups currently work really rely on custom minimum size, so it should work. Tho it's been a week, I'll check and try tomorrow to see if it will work.

Reply:
So... in terms of end result working meaning it goes to the proper size at the end of the animation; it works. But with how the new way it sizes itself with the panel container and all that the animation looks objectively horrible compared to how it was before any changes. I'm not really sure there is a way to make the size changing version of the animation look better really without potentially just over complicating the whole way of sizing popup menus again.

[Here's an example of the issue on Imgur](https://imgur.com/a/eDpSaxa)

Reply:
Okay, yeah that looks bad and was probably the reason why the size of the popups was overcomplicated. Let's just stick to the scale animation then as that allows the natural container sizing with the popup which solves a lot more headaches than it causes by having a bit worse animation.

Reply:
Can I see a gif showing how this animation looks in action? My local working git branch is a giant mess of uncommitted changes right now is so I can't easily checkout branches.

Reply:
@Athriq Here. 
[Scale-based animation example on Imgur](https://imgur.com/a/5bGWAON)

Reply:
Well, yeah that looks good enough to me.